app/services/expired_license_allower.rb

```diff
@@ -48,8 +48,9 @@ def allowable_expired_license?
     !!(
       only_error_was_document_expired? &&
         response.pii_from_doc[:state_id_expiration] &&
-        (state_id_expiration = Date.parse(response.pii_from_doc[:state_id_expiration])) &&
-        state_id_expiration >= IdentityConfig.store.proofing_expired_license_after
+        # Can switch to Date.parse after next deploy
+        (state_id_exp = DateParser.parse_legacy(response.pii_from_doc[:state_id_expiration])) &&
+        state_id_exp >= IdentityConfig.store.proofing_expired_license_after
```

Comment:
@zachmargolis Should we revisit this re: "next deploy" comment?

(Noticed while auditing similar code)

Reply:
https://github.com/18F/identity-idp/pull/6021